### PR TITLE
salut: init at unstable-2022-12-17

### LINKED ISSUE #265554: Package request: salut
**Project description**

A sleek notification daemon for Wayland compositors.

**Metadata**

* homepage URL: https://gitlab.com/snakedye/salut/-/wikis/Home
* source URL: https://gitlab.com/snakedye/salut
* license: mpl
* platforms: unix, linux, darwin, ...

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

Adds salut, a notification daemon for Wayland.

Fixes #265554.

Minimal config can be copied from the repo into `~/.config/salut/config.ini`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/sa/salut/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  ];
+
+  buildInputs = [
+    fontconfig
+    libxkbcommon
+    wayland
+  ];
+
+  meta = {
+    description = "A sleek notification daemon for Wayland";
```
Comment (FliegendeWurst) [inline pkgs/by-name/sa/salut/package.nix:42]:
```suggestion
    description = "Sleek notification daemon for Wayland";
```

**Review by FliegendeWurst — APPROVED** [review]
Result of `nixpkgs-review pr 265564` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>salut</li>
  </ul>
</details>

File: pkgs/by-name/sa/salut/package.nix (hunk 1 above)
Comment (eclairevoyant) [reply pkgs/by-name/sa/salut/package.nix:42]: <https://nixos.org/manual/nixpkgs/unstable/#var-meta-description>

> Don’t include a period at the end. Don’t include newline characters. Capitalise the first character. For brevity, don’t repeat the name of package — just describe what it does.
> 
> Wrong: "libpng is a library that allows you to decode PNG images."
>
> Right: ___"A library___ for decoding PNG images"

Comment (FliegendeWurst) [reply pkgs/by-name/sa/salut/package.nix:42]: I guess it depends on how you read it:
- `pname` is `meta.description`
- `pname` is a/an `meta.description`

Personally I prefer the latter, but I guess it was always intended to be the former: 79d3e0926d68bb760f1a9f28c862c38e6cd8cc45

**Comment by KiaraGrouwstra** [comment]
the package for me builds successfully [as well](https://codeberg.org/kiara/cfg/commit/800e7812c67f4f1f12ac97eb43cf437db516d382).
i'll have to admit i'm new to this package, but using the default config and a test message (`$ notify-send vanilla "Sunset Rollercoaster"`) i'm unfortunately still running into the following error: `GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.Notifications was not provided by any .service files`

**Comment by eclairevoyant** [comment]
I tested and it works fine for me - you do have to ensure to disable ny other notifications handlers and then start salut.

**Pull request merged**